File: DSA/Ananya_Gambhir/Day2_ques1.java
import java.util.*;
public class Day2_ques1 {

    public static void isCancel(int n,int time[]){
        int onTime=0;

        for (int i=0;i<time.length; i++) {
            if (time[i] <= 0) {
                onTime++;
            }
        }

        if(onTime<n){
            System.out.println("YES! The class is cancelled.");
        }
        else{
            System.out.println("NO! The class is not cancelled.");
        }
    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
            
        System.out.println("Enter the threshold: ");
        int n=sc.nextInt();
        System.out.println("Enter the no of students: ");
        int m=sc.nextInt();
        int time[]=new int[m];
        System.out.println("Enter the arrival times of students: ");
        for (int i = 0; i < m; i++) {
            time[i] = sc.nextInt();
        }

        isCancel(n, time);
        sc.close();
    }
}

File: DSA/Umang-Singhal/Day2_Ques1.c
#include <stdio.h>
int main() {
    int n; 
    printf("Enter the number of students: ");
    scanf("%d", &n);
    int threshold; 
    printf("Enter the threshold number of students: ");
    scanf("%d", &threshold);
    int arrival_times[n];
    printf("Enter the arrival times of the students: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arrival_times[i]);
    }
    int on_time_count = 0; 
    for (int i = 0; i < n; i++) {
        if (arrival_times[i] <= 0) {
            on_time_count++;
        }
    }
        if (on_time_count < threshold) {
        printf("YES\n");
    } else {
        printf("NO\n");
    }
    return 0;
}
